radicale/__init__.py

```diff
@@ -425,6 +433,13 @@ def response(status, headers=(), answer=None):
                     environ, base_prefix, path, user)
             except socket.timeout:
                 return response(*REQUEST_TIMEOUT)
+            except vobject.base.VObjectError as e:
+                return response(*UNPROCESSABLE_ENTITY(e))
```

Comment:
From the RFC:
> syntax of the request entity is correct

This also catches syntax errors. 400 is more appropriate.

> And if (for whatever reason) vObjects are parsed in another method

But the 4xx errors are indicating client errors. If a vobject from the storage can't be parsed, it's not a client error.